src/backends/smtlib2.rs

```diff
@@ -55,14 +56,40 @@ pub trait SMTProc {
         let mut bytes_read = [0; 2048];
         let mut s = String::new();
         let solver = self.pipe();
+
+
         if let Some(ref mut stdout) = solver.stdout.as_mut() {
-            loop {
-                let n = stdout.read(&mut bytes_read).unwrap();
-                s = format!("{}{}",
-                            s,
-                            String::from_utf8(bytes_read[0..n].to_vec()).unwrap());
-                if n < 2048 {
-                    break;
+            let mut flag = 0;
+            while flag!=3 {
+
+                for (i,c) in stdout.bytes().enumerate() {
+                    let chr = c.unwrap() as char;
+                    s.push(chr);
+                    match chr {
+                        's' => {
+                            if flag==0 {
+                                flag=1;
+                            } else {
+                                flag=0;
+                            }
+                        },
+                        'a' => {
+                            if flag==1 {
+                                flag=2;
+                            } else {
+                                flag=0;
+                            }
+                        },
+                        't' => {
+                            if flag==2 {
+                                flag=3;
+                                return s;
+                            } else {
+                                flag=0;
+                            }
+                        },
+                        _ => { flag=0; }
+                    }
```

Comment:
Because it was the only way it would work for me. It captures all the output including the error (for me)